src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_Range.cs

```diff
@@ -0,0 +1,173 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Diagnostics;
+using System.Collections.Immutable;
+using Microsoft.CodeAnalysis.CSharp.Symbols;
+using Microsoft.CodeAnalysis.PooledObjects;
+using Roslyn.Utilities;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Microsoft.CodeAnalysis.CSharp
+{
+    internal sealed partial class LocalRewriter
+    {
+        public override BoundNode VisitRangeExpression(BoundRangeExpression node)
+        {
+            Debug.Assert(node != null);
+
+            if (node.LeftOperand is null)
+            {
+                if (node.RightOperand is null)
+                {
+                    // Case 1:              ".."
+                    // translates to:       Range.All()
+
+                    MethodSymbol method = (MethodSymbol)_compilation.GetWellKnownTypeMember(WellKnownMember.System_Range__All);
+                    return MakeCall(node.Syntax, rewrittenReceiver: null, method, ImmutableArray<BoundExpression>.Empty, node.Type);
+                }
+
+                BoundExpression rightOperand = VisitExpression(node.RightOperand);
+                rightOperand = NullableAlwaysHasValue(rightOperand) ?? rightOperand;
+
+                if (node.RightOperand.Type.IsNullableType())
+                {
+                    // Case 2:              "..nullableRight"
+                    // translates to:       nullableRight.HasValue ? new Nullable(Range.ToEnd(nullableRight.GetValueOrDefault())) : default
+
+                    MethodSymbol method = (MethodSymbol)_compilation.GetWellKnownTypeMember(WellKnownMember.System_Range__ToEnd);
+                    return LiftRangeExpression(node.Syntax, node.Type, method, rightOperand);
+                }
+                else
+                {
+                    // Case 3:              "..right"
+                    // translates to:       Range.ToEnd(right)
+
+                    MethodSymbol method = (MethodSymbol)_compilation.GetWellKnownTypeMember(WellKnownMember.System_Range__ToEnd);
+                    ImmutableArray<BoundExpression> arguments = ImmutableArray.Create(rightOperand);
+                    return MakeCall(node.Syntax, rewrittenReceiver: null, method, arguments, node.Type);
```

Comment:
>node.Type [](start = 93, length = 9)

Wouldn't this be a wrong type for the call when ```node.Type``` is nullable? It feels like we should use method's return type here and then make nullable out of this call if needed.

Reply:
`node.Type` would never be nullable here. I'll add an assert.

---
In reply to: [185366478](https://github.com/dotnet/roslyn/pull/26405#discussion_r185366478) [](ancestors = 185366478)

Reply:
> node.Type would never be nullable here. I'll change it to the method's return type anyways.

I don't thin this is correct assumption. We might have erased nullability from operand, but that doesn't change the type of the result for the entire node from being nullable.

---
In reply to: [185369721](https://github.com/dotnet/roslyn/pull/26405#discussion_r185369721) [](ancestors = 185369721,185366478)

Reply:
Thanks for clarifying!

---
In reply to: [185485969](https://github.com/dotnet/roslyn/pull/26405#discussion_r185485969) [](ancestors = 185485969,185369721,185366478)